astropy/io/misc/pyarrow/tests/test_csv.py

```diff
@@ -0,0 +1,605 @@
+import contextlib
+import io
+import tempfile
+import textwrap
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from astropy.table import Table
+from astropy.utils.compat.optional_deps import HAS_PYARROW
+
+if HAS_PYARROW:
+    import pyarrow as pa
+else:
+    pytest.skip("pyarrow is not available")
+
+
+def check_tables_equal(t1, t2):
+    """Check that two tables are equal.
+
+    This function checks that the two tables have the same column names,
+    lengths, and that the data in each column is equal. It also checks
+    that the masks (if any) are equal.
+    """
+    assert t1.colnames == t2.colnames
+    assert len(t1) == len(t2)
+    for col1, col2 in zip(t1.itercols(), t2.itercols()):
+        assert col1.name == col2.name
+        assert col1.dtype == col2.dtype
+        assert np.array_equal(col1, col2)
```

Comment:
I'd use `numpy.testing`'s `assert_array_equal` - gives nicer error message on failure, pointing out where the discrepancy is.

Reply:
Done